packages/frontend-core/src/utils/components.ts

```diff
@@ -0,0 +1,26 @@
+import { ComponentDefinition, ComponentSetting } from "@budibase/types"
+
+/**
+ * Gets the definition of this component's settings from the manifest
+ */
+export const getSettingsDefinition = (
+  definition: ComponentDefinition
+): ComponentSetting[] => {
+  if (!definition) {
+    return []
+  }
+  let settings: ComponentSetting[] = []
+  definition.settings?.forEach(setting => {
+    if (setting.section) {
+      settings = settings.concat(
+        (setting.settings || [])?.map(childSetting => ({
+          ...childSetting,
+          sectionDependsOn: setting.dependsOn,
+        }))
+      )
+    } else {
+      settings.push(setting)
+    }
+  })
+  return settings
+}
```

Comment:
We have `componentStore.getComponentSettings` already - does this basically just do the same thing? That function gets all settings for a certain component type and flattens them into a single array, accounting for sections.

Reply:
This does not work, as it flattens the list completely. We have some settings, such as `Date column` that depends on a higher level dependsOn. So we need to iterate trough them instead to be able to see if it's actually required.
To be clear, this code from here is the same that was being used within the client, I just extracted it